### DISCUSSION TITLE
Export a realm from a keycloak devservice?

### DISCUSSION THREAD

**Discussion by HerrDerb** [quarkusio/quarkus#31512] [Q&A] [answered]
I would like to export a realm from a running keylcoak devservice.
There I seem to have a chicken/egg problem paired with non-persistance.

In order to create  the realm I would like to export, I need to start the devservice and make my configuration.
To export that realm, I would need to stop the keycloak server and start the export process. As I there is no way to stop a keycloak server on the current docker image, I could just restart the docker image with a new entry point command.

But this would loos the whole configuration as the dev-file is not persisted, and I see no way to bind a volume to my filesystem with the devservice.

How could I export a realm from a keycloak devservice, if at all?

**Comment by sberyozkin** [comment, marked as answer]
@HerrDerb I've meant to discuss the options with Pedro @pedroigor, but so far the issue to follow is #30097. If we can have an option to fetch with Keycloak admin API as suggested by @geoand then it would be best.
FYI, you can export the realm using Keycloak UI, I'm assuming you mean that it does not export all the information, such as users ?

**Reply by HerrDerb** [reply]
Yes, I would like to be able to implement the export as a build task to extract the realm.json . Ah nice, I'll hang on to the issue then (y)
